src/Assignments/ProblemIframe.tsx

```diff
@@ -67,8 +69,32 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
         // when problem changes, reset lastsubmitted and lastsaved
         setLastSubmittedAt?.(null);
         setLastSavedAt?.(null);
+        setLastSubmission({});
+        updateSubmitActive(false);
     }, [problem.id]);
 
+    const updateSubmitActive = (incomingIsClean: boolean | undefined) => {
```

Comment:
Method name needs to be renamed
updateSubmitActive true seems like you are saying you want to make submit active
However the parameter is the negative meaning (which you don't see without a good idea)

a clean object should be disabled

Reply:
Let's change to isActive

Reply:
remove "| undefined"